diff --git a/src/AppBundle/Repository/ProductRepository.php b/src/AppBundle/Repository/ProductRepository.php
index 14dfa8b..1bcc55f 100644
--- a/src/AppBundle/Repository/ProductRepository.php
+++ b/src/AppBundle/Repository/ProductRepository.php
@@ -45,15 +45,15 @@ class ProductRepository
 
     public function insert(Product $product)
     {
-        $stmt = $this->pdo->prepare('
-            INSERT INTO product (name, description, price, created_at)
-            VALUES (:name, :description, :price, :createdAt)
-        ');
-        $stmt->bindValue(':name', $product->getName());
-        $stmt->bindValue(':description', $product->getDescription());
-        $stmt->bindValue(':price', $product->getPrice());
-        $stmt->bindValue(':createdAt', date('Y-m-d H:i:s'));
-
-        $stmt->execute();
+        $data = [
+            'name' => $product->getName(),
+            'description' => $product->getDescription(),
+            'price' => $product->getPrice(),
+            'createdAt' => date('Y-m-d H:i:s')
+        ];
+
+        $this->client->post('/products.json', [
+            'body' => json_encode($data)
+        ]);
     }
 }
\ No newline at end of file
