diff --git a/src/ApiBundle/DependencyInjection/ApiExtension.php b/src/ApiBundle/DependencyInjection/ApiExtension.php
new file mode 100644
index 0000000..6fe5b92
--- /dev/null
+++ b/src/ApiBundle/DependencyInjection/ApiExtension.php
@@ -0,0 +1,20 @@
+<?php
+
+namespace ApiBundle\DependencyInjection;
+
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+class ApiExtension extends Extension
+{
+    public function load(array $configs, ContainerBuilder $container)
+    {
+        $loader = new YamlFileLoader(
+            $container,
+            new FileLocator(__DIR__.'/../Resources/config')
+        );
+        $loader->load('services.yml');
+    }
+}
\ No newline at end of file
