diff --git a/src/AppBundle/Controller/ProductController.php b/src/AppBundle/Controller/ProductController.php
index 3e00f94..669f455 100644
--- a/src/AppBundle/Controller/ProductController.php
+++ b/src/AppBundle/Controller/ProductController.php
@@ -7,6 +7,7 @@ use AppBundle\Model\Product;
 use AppBundle\Repository\ProductRepository;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
 
 class ProductController extends Controller
 {
@@ -25,10 +26,15 @@ class ProductController extends Controller
     /**
      * @Route("/products/new", name="product_new")
      */
-    public function newAction()
+    public function newAction(Request $request)
     {
         $form = $this->createForm(ProductForm::class);
 
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            dump($form->getData());die;
+        }
+
         return $this->render('product/new.html.twig', [
             'form' => $form->createView()
         ]);
