diff --git a/app/Resources/views/product/index.html.twig b/app/Resources/views/product/index.html.twig
index 240ed9e..a8d2add 100644
--- a/app/Resources/views/product/index.html.twig
+++ b/app/Resources/views/product/index.html.twig
@@ -1,12 +1,12 @@
 {% extends 'base.html.twig' %}
 
 {% block body %}
-    {% for product in rows %}
+    {% for product in products %}
         <div>
             <h3>{{ product.name }}</h3>
             {{ product.description }}
             {{ product.price }}
-            {{ product.created_at }}
+            {{ product.createdAt|date('Y-m-d') }}
         </div>
     {% endfor %}
 {% endblock %}
\ No newline at end of file
diff --git a/src/AppBundle/Controller/ProductController.php b/src/AppBundle/Controller/ProductController.php
index c949458..5894aa6 100644
--- a/src/AppBundle/Controller/ProductController.php
+++ b/src/AppBundle/Controller/ProductController.php
@@ -2,6 +2,7 @@
 
 namespace AppBundle\Controller;
 
+use AppBundle\Model\Product;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 
@@ -16,8 +17,20 @@ class ProductController extends Controller
         $db->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
         $result = $db->query('SELECT * FROM product');
 
+        $products = [];
+        foreach ($result as $row) {
+            $product = new Product();
+            $product->setId($row['id']);
+            $product->setName($row['name']);
+            $product->setDescription($row['description']);
+            $product->setPrice($row['price']);
+            $product->setCreatedAt(\DateTime::createFromFormat('Y-m-d H:i:s', $row['created_at']));
+
+            $products[] = $product;
+        }
+
         return $this->render('product/index.html.twig', [
-            'rows' => $result
+            'products' => $products
         ]);
     }
 }
\ No newline at end of file
diff --git a/src/AppBundle/Model/Product.php b/src/AppBundle/Model/Product.php
new file mode 100644
index 0000000..68a20f1
--- /dev/null
+++ b/src/AppBundle/Model/Product.php
@@ -0,0 +1,66 @@
+<?php
+
+namespace AppBundle\Model;
+
+class Product
+{
+    private $id;
+
+    private $name;
+
+    private $description;
+
+    private $price;
+
+    private $createdAt;
+
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    public function setId($id)
+    {
+        $this->id = $id;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    public function getDescription()
+    {
+        return $this->description;
+    }
+
+    public function setDescription($description)
+    {
+        $this->description = $description;
+    }
+
+    public function getPrice()
+    {
+        return $this->price;
+    }
+
+    public function setPrice($price)
+    {
+        $this->price = $price;
+    }
+
+    public function getCreatedAt()
+    {
+        return $this->createdAt;
+    }
+
+    public function setCreatedAt($createdAt)
+    {
+        $this->createdAt = $createdAt;
+    }
+}
