diff --git a/app/config/security.yml b/app/config/security.yml
index a5bdd1f..7e0e97a 100644
--- a/app/config/security.yml
+++ b/app/config/security.yml
@@ -23,5 +23,8 @@ security:
                 authenticators:
                     - query_string_authenticator
 
+            logout:
+                path: /logout
+
             # form_login: ~
             # http://symfony.com/doc/current/cookbook/security/form_login_setup.html
diff --git a/src/AppBundle/Controller/SecurityController.php b/src/AppBundle/Controller/SecurityController.php
index 0db9479..1d7229e 100644
--- a/src/AppBundle/Controller/SecurityController.php
+++ b/src/AppBundle/Controller/SecurityController.php
@@ -15,4 +15,12 @@ class SecurityController extends Controller
     {
         // should not be executed
     }
+
+    /**
+     * @Route("/logout")
+     */
+    public function logoutAction()
+    {
+        // should not be executed
+    }
 }
