diff --git a/src/AppBundle/Model/User.php b/src/AppBundle/Model/User.php
index 5c95f87..84aaea9 100644
--- a/src/AppBundle/Model/User.php
+++ b/src/AppBundle/Model/User.php
@@ -6,28 +6,32 @@ use Symfony\Component\Security\Core\User\UserInterface;
 
 class User implements UserInterface
 {
+    private $username;
+
+    public function __construct($username)
+    {
+        $this->username = $username;
+    }
+
     public function getUsername()
     {
-        // TODO: Implement getUsername() method.
+        return $this->username;
     }
 
     public function getRoles()
     {
-        // TODO: Implement getUsername() method.
+        return ['ROLE_USER'];
     }
 
     public function getPassword()
     {
-        // TODO: Implement getPassword() method.
     }
 
     public function getSalt()
     {
-        // TODO: Implement getSalt() method.
     }
 
     public function eraseCredentials()
     {
-        // TODO: Implement eraseCredentials() method.
     }
 }
